src/Coordination/Changelog.cpp

```diff
@@ -157,21 +157,24 @@ class ChangelogWriter
         assert(file_buffer && current_file_description);
 
         assert(record.header.index - getStartIndex() <= current_file_description->expectedEntriesCountInLog());
-        const bool log_is_complete = record.header.index - getStartIndex() == current_file_description->expectedEntriesCountInLog();
-
-        if (log_is_complete)
-            rotate(record.header.index);
-
-        // writing at least 1 log is requirement - we don't want empty log files
-        // we use count() that can be unreliable for more complex WriteBuffers, so we should be careful if we change the type of it in the future
-        const bool log_too_big = record.header.index != getStartIndex() && log_file_settings.max_size != 0
-            && initial_file_size + file_buffer->count() > log_file_settings.max_size;
-
-        if (log_too_big)
+        if (const bool log_is_complete = record.header.index - getStartIndex() == current_file_description->expectedEntriesCountInLog();
```

Comment:
It is hard to parse. Why don't just write like this?
`if (record.header.index - getStartIndex() == current_file_description->expectedEntriesCountInLog())`
We could introduce a function with explanation name for this expression.

Reply:
This was autopilot change, I saw a variable that is used for if only, and I just turned it into `if-with-initializer` without noticing that the variable is a bool and can be written as you said.
I changed to your suggestion + added a comment to keep the clarity. Thanks for noticing!